pfcpiface/messages.go

```diff
@@ -632,3 +665,72 @@ func (pc *PFCPConn) generateAssociationRequest(n4LocalIP string, n3ip string, n4
 		log.Fatalln("Unable to transmit association setup request ", err)
 	}
 }
+
+func readReportNotification(rn <-chan uint64, pfcpConn *PFCPConn,
+	udpConn *net.UDPConn, udpAddr net.Addr) {
+	log.Println("read report notification start")
+	for {
+		select {
+		case fseid := <-rn:
+			handleDigestReport(fseid, pfcpConn, udpConn, udpAddr)
+
+		default:
+			time.Sleep(2 * time.Second)
+		}
+	}
+}
+
+func handleDigestReport(fseid uint64,
+	pfcpConn *PFCPConn,
+	udpConn *net.UDPConn,
+	udpAddr net.Addr) {
+	session, ok := pfcpConn.mgr.sessions[fseid]
+	if !ok {
+		log.Println("No session found for fseid : ", fseid)
+		return
+	}
+
+	if session.notifyCount != 0 {
+		return
+	}
+
+	session.notifyCount++
```

Comment:
where do we decrement this? should be in the response handling?

Reply:
do we need to get a lock for this?

Reply:
done